lib/elixir/lib/stream.ex

```diff
@@ -1404,6 +1401,22 @@ defmodule Stream do
     do: %{lazy | funs: [fun | funs], accs: [acc | accs], done: done}
   defp lazy(enum, acc, fun, done),
     do: %Stream{enum: enum, funs: [fun], accs: [acc], done: done}
+
+  defp do_interval({:first_time, interval}) do
+    { 0, { 1, now_in_ms(), interval }}
+  end
+
+  defp do_interval({count, previous_time, interval}) do
+    then = previous_time + interval
+    now  = now_in_ms()
+
+    if then > now, do: Process.sleep(then - now)
+
+    {count, { count + 1, then, interval }}
+  end
+    
+  defp now_in_ms(),
```

Comment:
Also considering that this name contains unit in it and what type of time is that, if we wanted to change the implementation, it means that it should be done in those two place as well.